delegate: gate estimated_row_count on SHOW TABLES behind cluster setting

otan opened the pull request:
Refs: #58189

Release note (sql change): Introduced a cluster setting
sql.show_tables.estimated_row_count.enabled, which defaults to true. If
false, estimated_row_count will not display on SHOW TABLES which
improves performance.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/60282)
<!-- Reviewable:end -->

rafiss reviewed (approved):
excelente

otan committed:
delegate: gate estimated_row_count on SHOW TABLES behind cluster setting

Release note (sql change): Introduced a cluster setting
sql.show_tables.estimated_row_count.enabled, which defaults to true. If
false, estimated_row_count will not display on SHOW TABLES which
improves performance.

otan commented:
bors r=rafiss

otan commented:
bors r-

otan commented:
bors r=rafiss
